packages/react-components/src/queries/useLatestAssetPropertyValues/useLatestAssetPropertyValues.ts

```diff
@@ -0,0 +1,184 @@
+import { useMemo } from 'react';
+import { QueryFunctionContext, useQueries } from '@tanstack/react-query';
+import invariant from 'tiny-invariant';
+
+import { useIoTSiteWiseClient } from '../../hooks/requestFunctions/useIoTSiteWiseClient';
+import { queryClient } from '../queryClient';
+import { isAssetId, isPropertyAlias, isPropertyId } from '../predicates';
+
+import {
+  LatestAssetPropertyValueRequest,
+  QueryFnClient,
+  UseLatestAssetPropertyValuesOptions,
+} from './types';
+import { LatestAssetPropertyValueKeyFactory } from './latestAssetPropertyValueKeyFactory';
+import { useSyncLatestAssetPropertyValueQueries } from './sync';
+import {
+  GetLatestAssetPropertyValueRequest,
+  LatestAssetPropertyValueBatcher,
+} from './requestExecution';
+
+const DEFAULT_REFRESH_RATE = 5000;
+
+const requestIsValid = ({
+  assetId,
+  propertyId,
+  propertyAlias,
+}: LatestAssetPropertyValueRequest) =>
+  (isAssetId(assetId) && isPropertyId(propertyId)) ||
+  isPropertyAlias(propertyAlias);
+
+const clientIsValid = ({
+  getAssetPropertyValue,
+  batchGetAssetPropertyValue,
+}: QueryFnClient = {}) =>
+  getAssetPropertyValue != null || batchGetAssetPropertyValue != null;
+
+/**
+ * useLatestAssetPropertyValues is a hook used to call
+ * getAssetPropertyValue or batchGetAssetPropertyValue
+ * on a list of GetAssetPropertyValueRequest requests
+ *
+ * @param iotSiteWiseClient is an AWS SDK IoT SiteWise client
+ * @param enabled will manually disable the hook
+ * @param requests list of GetAssetPropertyValueRequest
+ * @param refreshRate refresh rate for GetAssetPropertyValueRequest in ms
+ * @param batchSechedulerTimeout configures the timeout window in ms to accept
+ * batch requests https://github.com/graphql/dataloader?tab=readme-ov-file#batch-scheduling
+ * (Note: can only be set once.)
+ * - defaults to 100ms
+ * @param batchSize configures the timeout window to accept
+ * (Note: can only be set once.)
+ * - defaults to 128
+ *
+ * @returns a list of QueryResults with GetAssetPropertyValueResponse data.
+ */
+export const useLatestAssetPropertyValues = ({
+  iotSiteWiseClient,
+  enabled = true,
+  requests = [],
+  refreshRate = DEFAULT_REFRESH_RATE,
+  batchSechedulerTimeout,
+  batchSize,
+}: UseLatestAssetPropertyValuesOptions) => {
+  /**
+   * A single enabled flag for all useLatestAssetPropertyValues
+   * with the same refreshRate. This will toggle the queries
+   * off / on in a setTimeout so that the executions all
+   * happen in the same task and the resulting refetchInterval
+   * executes within the batch timeout window.
+   */
+  const syncFlag = useSyncLatestAssetPropertyValueQueries({
+    enabled,
+    refreshRate,
+  });
+
+  const { getAssetPropertyValue, batchGetAssetPropertyValue } =
+    useIoTSiteWiseClient({ iotSiteWiseClient });
+
+  const queries = useMemo(() => {
+    return requests.map((request) => {
+      return {
+        enabled:
+          syncFlag &&
+          enabled &&
+          clientIsValid({
```

Comment:
Why do you need to validate the input here?

Reply:
The hook should only execute the request function if the inputs are valid. If there are no clients or the request is malformed, the hook is disabled.

Reply:
Oh got it there isn't an `isEnabled()` function, it's inline here